<?xml version='1.0' encoding='UTF-8'?>
<root version="2.3.8">
 <tabbed_widget name="Main Window" parent="main_window">
  <Tab containers="1" tab_name="tab1">
   <Container>
    <DockSplitter orientation="-" count="2" sizes="0.500772;0.499228">
     <DockSplitter orientation="|" count="3" sizes="0.333045;0.33391;0.333045">
      <DockArea name="Velocity Controller Info">
       <plot mode="TimeSeries" style="Lines">
        <range bottom="-0.009500" right="26.268925" top="0.389492" left="0.000000"/>
        <limitY/>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.3" color="#1ac938">
         <transform name="Scale/Offset" alias="FF">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.14" color="#ff7f0e">
         <transform name="Scale/Offset" alias="FF+FB">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.19" color="#f14cc1">
         <transform name="Scale/Offset" alias="FB-P">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.20" color="#9467bd">
         <transform name="Scale/Offset" alias="FB-I">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.21" color="#17becf">
         <transform name="Scale/Offset" alias="FB-D">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.15" color="#bcbd22">
         <transform name="Scale/Offset" alias="acc filtered">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.16" color="#1f77b4">
         <transform name="Scale/Offset" alias="jerk filtered">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.17" color="#d62728">
         <transform name="Scale/Offset" alias="slope added">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.18" color="#1ac938">
         <transform name="Scale/Offset" alias="final">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.25" color="#ff7f0e">
         <transform name="Scale/Offset" alias="calculated">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
       </plot>
      </DockArea>
      <DockArea name="State">
       <plot mode="TimeSeries" style="Lines">
        <range bottom="0.900000" right="26.268925" top="1.100000" left="0.000000"/>
        <limitY/>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.13" color="#d62728">
         <transform name="Scale/Offset" alias="state 0:Init 1:Ctrl 2:Stopped 3:SmoothStop 4:Emergency 5:Error">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
       </plot>
      </DockArea>
      <DockArea name="Stop Dist">
       <plot mode="TimeSeries" style="Lines">
        <range bottom="-2.402500" right="26.261692" top="-2.297500" left="0.414630"/>
        <limitY/>
        <curve name="/planning/scenario_planning/motion_velocity_optimizer/distance_to_stopline/data" color="#1f77b4">
         <transform name="Scale/Offset" alias="stop dist">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter orientation="|" count="3" sizes="0.332181;0.335638;0.332181">
      <DockArea name="Pacmod">
       <plot mode="TimeSeries" style="Lines">
        <range bottom="0.000000" right="1.000000" top="1.000000" left="0.000000"/>
        <limitY/>
       </plot>
      </DockArea>
      <DockArea name="Velocity">
       <plot mode="TimeSeries" style="Lines">
        <range bottom="-0.025000" right="26.280226" top="1.025000" left="0.000000"/>
        <limitY/>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.2" color="#17becf">
         <transform name="Scale/Offset" alias="target">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.1" color="#9467bd">
         <transform name="Scale/Offset" alias="current">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.4" color="#bcbd22">
         <transform name="Scale/Offset" alias="closest">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
        <curve name="/control/trajectory_follower/velocity_controller/debug_values/data.24" color="#1f77b4">
         <transform name="Scale/Offset" alias="predicted">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
        <curve name="/vehicle/status/control_mode/data" color="#d62728">
         <transform name="Scale/Offset" alias="Control Mode">
          <options value_scale="1.0" time_offset="0" value_offset="0"/>
         </transform>
        </curve>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot mode="TimeSeries" style="Lines">
        <range bottom="0.000000" right="1.000000" top="1.000000" left="0.000000"/>
        <limitY/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <currentTabIndex index="0"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="1"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <default time_axis=""/>
  </plugin>
  <plugin ID="DataLoad ROS2 bags">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="99"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="MQTT Subscriber"/>
  <plugin ID="ROS2 Topic Subscriber">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="false"/>
   <max_array_size value="true"/>
   <selected_topics>
    <topic name="/control/trajectory_follower/velocity_controller/debug_values"/>
    <topic name="/planning/scenario_planning/motion_velocity_optimizer/distance_to_stopline"/>
    <topic name="/vehicle/status/control_mode"/>
   </selected_topics>
  </plugin>
  <plugin ID="UDP Server"/>
  <plugin ID="WebSocket Server"/>
  <plugin ID="ZMQ Subscriber"/>
  <plugin ID="ROS2 Topic Re-Publisher" status="idle"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS2 Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets/>
 <!-- - - - - - - - - - - - - - - -->
</root>

